### PR TITLE
(POC) Add devfile v2 outerloop flow to console

### PR DESCRIPTION
**Pull request by reginapizza**
This PR is to implement the `Import from Devfile` flow.

 UX mocks for this flow can be found [here](https://docs.google.com/document/d/1f8a5gD-KOwW-nZO1LkDlT00H1JONlV8OcXqlJXqV8o4/edit?pli=1#heading=h.lci028wi43sy)

Changes made and what is left to do is documented [here](https://docs.google.com/document/d/1BCz8Ne5DKTMhgOHjX7G83j8PKq_t8iF4IB05Sfk8_Hk/edit).

Full demo video can be found [here](https://drive.google.com/file/d/1uLzDibVZlkMqbjtKkho04e8k2-Ns5A2W/view?usp=sharing).

@Shraddhak22 @jaideepr97 @ranakan19 @wtam2018 @mitchharpur

### CODE REVIEW

**Comment by andrewballantyne** [comment]
Thanks for opening up this PR @reginapizza - I'll look through and see what you have come up with. Would you mind updating the description with a couple screenshots / gifs of the work you have done so it's clearer at a glance?

/hold
/assign

Also a little bookkeeping - Master is currently pointing at 4.6, and this is a 4.7 implementation. So just going to put a cautionary hold on this PR while we review the changes and discuss things.

**Comment by reginapizza** [comment]
@andrewballantyne PR description has been updated with screenshots!

My apologies about not understanding the versioning, still getting used to things here... should I update from `master` to `relesease-4.7` then?

**Comment by reginapizza** [comment]
Just a quick update: things are working on the UI side for the most part, the last big puzzle piece to complete the flow is getting the devfile/parser library to work. The diagram below shows how the communication works between the app and the library; in place of the devfile/parser library we are using mocked data in the backend. 

importFromDevfileFlowchart

Please see [this doc](https://docs.google.com/document/d/1BCz8Ne5DKTMhgOHjX7G83j8PKq_t8iF4IB05Sfk8_Hk/edit) for a more in-depth status update, including items that are still on the to-do list.

**Review by serenamarie125 — CHANGES_REQUESTED** [review]
Please add the "Dev Preview" badge on the Import from devfile form page.  Thanks!

**Comment by reginapizza** [comment]
> Please add the "Dev Preview" badge on the Import from devfile form page. Thanks!

@serenamarie125  The images in the original description were outdated, the badge is currently there though. I've updated the description to include a full demo video that accurately reflects the code here now.

**Comment by serenamarie125** [comment]
> @serenamarie125 The images in the original description were outdated, the badge is currently there though. I've updated the description to include a full demo video that accurately reflects the code here now.

Looks great, thanks @reginapizza

**Comment by jaideepr97** [comment]
> @reginapizza Are we going to want to get this in before the backend is ready. Based on the meeting yesterday it sounded like they are moving the right direction but are not yet ready. I think we may be able to get this into the UI and swing back with a go update to add the proper connectivity to the devfile lib.
> 
> Thoughts?

Sounds good to me @andrewballantyne

**Comment by maysunfaisal** [comment]
FYI, the devfile library PR is up for review at https://github.com/devfile/library/pull/44

**Comment by andrewballantyne** [comment]
I'll look to take over this PR going into next sprint. Thanks @maysunfaisal !

File: frontend/packages/dev-console/src/components/import/git/GitSection.tsx [hunk 1]
```diff
@@ -157,6 +189,9 @@ const GitSection: React.FC<GitSectionProps> = ({ showSample, builderImages }) =>
     if (validated === ValidatedOptions.warning) {
       return 'URL is valid but cannot be reached. If this is a private repository, enter a source secret in Advanced Git Options';
     }
+    if (validated === ValidatedOptions.warning) {
+      return 'Valid URL but devfile build guidance not available.';
+    }
```
Comment (maysunfaisal) [inline frontend/packages/dev-console/src/components/import/git/GitSection.tsx:194]: Both these if conditions check for `warning`. Looks like the condition for not reachable should be an `error` looking at the logic above in this file

Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/import/git/GitSection.tsx:194]: Interesting, I'll have to validate the UI code once I move it over. I'm trying to close off the ticket I'm working on but clusters seem to be having some issues today. Good catch though.

**Comment by andrewballantyne** [comment]
/close

See #7299 for the continued work on Devfile addition to DevConsole.

**Pull request closed without merging**